jira-automation/migrations/sync-all-issues-from-jira-to-github.mjs

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env node
+
+import { Octokit } from "@octokit/rest";
+import {jiraToGitHub} from "../jira-to-github-processing.mjs"
+import { getIssue } from "../jira.mjs";
+import { updateGitHubIssue } from "../github.mjs";
+
+const token = process.env.GITHUB_TOKEN || process.env.GITHUB_REPO_TOKEN;
+
+
+// Create an Octokit instance with your GitHub personal access token
+const octokit = new Octokit({
+  auth: token
+});
+
+// Fetch all open issues from the repository
+async function getOpenIssues() {
+  const response = await octokit.paginate(octokit.issues.listForRepo, {
+    owner: "hmcts",
+    repo: "roadmap-platform-operations",
+    state: "open",
+  });
+
+  // return response;
+  return response;
+}
+
+async function run() {
+  const issues = await getOpenIssues();
+
+  for (const issue of issues) {
+
+    console.log('Processing', issue.title)
+    const regexResult = issue.body?.match(/DTSPO-\d+/)
+    if (!regexResult) {
+      // normally tech debt issues
+      console.log('No Jira issue found, skipping')
+      continue
+    }
+    const issueKey = regexResult[0]
+
+    console.log('Fetching Jira', issueKey);
+
+    let jiraIssue
+    try {
+      jiraIssue = await getIssue({ key: issueKey })
+    } catch (err) {
+      console.log("Error searching for issue in jira", issueKey, err)
+
+      process.exit(1)
```

Comment:
I take this will end the whole migration process. in the future we might implement  a `retry` mechanism here to continue the migration. sometimes it might be just a temporary network glitch

Reply:
could do but i'd rather add resiliency as we hit issues and fail fast. it's been ran a few times with no issues now

Reply:
I agree